src/dnd/DraggableController.js

```
@@ -0,0 +1,265 @@
+
+        // Translate sensorEvent coordinates to element under pointer.
+        const clientX = sensorEvent.clientX;
+        const clientY = sensorEvent.clientY;
+        const el = document.elementFromPoint(clientX, clientY);
+        if (!el) return;
+        const td = el.closest('td[data-r][data-c]');
+        if (!td || !this._refs.dropRoot.contains(td)) return;
+
+        // Extract address.
+        const row = parseInt(td.dataset.r, 10);
+        const col = parseInt(td.dataset.c, 10);
```

Biome (1.9.4)

[error] Use Number.parseInt instead of the equivalent global.

ES2015 moved some globals into the Number namespace for consistency.
Safe fix: Use Number.parseInt instead.


(lint/style/useNumberNamespace)